worlds/pokemon_rb/rom.py

```diff
@@ -0,0 +1,629 @@
+        for i in range(0, 3):
+            data[rom_addresses['Option_Boulders'] + (i * 3)] = 0x15
+    if self.world.extra_key_items[self.player].value:
+        for i in range(0, 4):
+            data[rom_addresses['Option_Rock_Tunnel_Extra_Items'] + (i * 3)] = 0x15
+    if self.world.old_man[self.player].value == 2:
+        data[rom_addresses['Option_Old_Man']] = 0x11
+        data[rom_addresses['Option_Old_Man_Lying']] = 0x15
+    money = str(self.world.starting_money[self.player].value)
+    while len(money) < 6:
+        money = "0" + money
+    data[rom_addresses["Starting_Money_High"]] = int(money[:2], 16)
+    data[rom_addresses["Starting_Money_Middle"]] = int(money[2:4], 16)
+    data[rom_addresses["Starting_Money_Low"]] = int(money[4:], 16)
+    data[rom_addresses["Text_Badges_Needed"]] = encode_text(
+        str(max(self.world.victory_road_condition[self.player].value,
+                self.world.elite_four_condition[self.player].value)))[0]
+    if self.world.badges_needed_for_hm_moves[self.player].value == 0:
+        for hm_move in poke_data.hm_moves:
+            write_bytes(data, bytearray([0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]),
+                        rom_addresses["HM_" + hm_move + "_Badge_a"])
+    elif self.extra_badges:
+        written_badges = {}
+        for hm_move, badge in self.extra_badges.items():
+            data[rom_addresses["HM_" + hm_move + "_Badge_b"]] = {"Boulder Badge": 0x47, "Cascade Badge": 0x4F,
+                                                                 "Thunder Badge": 0x57, "Rainbow Badge": 0x5F,
+                                                                 "Soul Badge": 0x67, "Marsh Badge": 0x6F,
+                                                                 "Volcano Badge": 0x77, "Earth Badge": 0x7F}[badge]
+            move_text = hm_move
+            if badge not in ["Marsh Badge", "Volcano Badge", "Earth Badge"]:
+                move_text = ", " + move_text
+            rom_address = rom_addresses["Badge_Text_" + badge.replace(" ", "_")]
+            if badge in written_badges:
+                rom_address += len(written_badges[badge])
+                move_text = ", " + move_text
+            write_bytes(data, encode_text(move_text.upper()), rom_address)
+            written_badges[badge] = move_text
+        for badge in ["Marsh Badge", "Volcano Badge", "Earth Badge"]:
+            if badge not in written_badges:
+                write_bytes(data, encode_text("Nothing"), rom_addresses["Badge_Text_" + badge.replace(" ", "_")])
+
+    chart = deepcopy(poke_data.type_chart)
+    if self.world.randomize_type_matchup_types[self.player].value == 1:
+        attacking_types = []
+        defending_types = []
+        for matchup in chart:
+            attacking_types.append(matchup[0])
+            defending_types.append(matchup[1])
+        self.world.random.shuffle(attacking_types)
+        self.world.random.shuffle(defending_types)
```

Comment:
```suggestion
        random.shuffle(attacking_types)
        random.shuffle(defending_types)
```
Can't use global random source here as it's run in a thread -> undefined result -> no seed reproducibilty